package/MDAnalysis/lib/qcprot.pyx

```diff
@@ -174,27 +176,25 @@ cdef extern from "math.h":
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
-def InnerProduct(np.ndarray[np.float64_t, ndim=1] A,
-                 np.ndarray[np.float64_t, ndim=2] coords1,
-                 np.ndarray[np.float64_t, ndim=2] coords2,
-                 int N,
-                 np.ndarray[np.float64_t, ndim=1] weight):
+cpdef cython.floating InnerProduct(cython.floating[:] A,
+                                   cython.floating[:, :] coords1,
+                                   cython.floating[:, :] coords2,
+                                   int N,
+                                   cython.floating[:] weight):
```

Comment:
I remember that a while back I tried to improve performance in SciPy by switching from the old NumPy buffer syntax, to memoryviews, and it actually caused a performance regression: https://github.com/scipy/scipy/issues/12379#issuecomment-645934588

Anyway, you may want to measure that you remain at least performance neutral, my experience with following that best practice hasn't historically been great, but hopefully the situation is better now.